src/process/mod.rs

```diff
@@ -719,12 +872,79 @@ impl Process {
 
         for line in reader.lines() {
             let line = line.map_err(|_| ProcError::Incomplete(Some(path.clone())))?;
-            vec.push(from_line(&line)?);
+            vec.push(MemoryMap::from_line(&line)?);
         }
 
         Ok(vec)
     }
 
+    /// Returns a list of currently mapped memory regions and verbose information about them,
+    /// such as memory consumption per mapping, based on the `/proc/pid/smaps` file.
+    ///
+    /// (since Linux 2.6.14 and requires CONFIG_PROG_PAGE_MONITOR)
+    pub fn smaps(&self) -> ProcResult<HashMap<MemoryMap, MemoryMapData>> {
+        let path = self.root.join("smaps");
+        let file = FileWrapper::open(&path)?;
+
+        let reader = BufReader::new(file);
+
+        let mut hashmap: HashMap<MemoryMap, MemoryMapData> = HashMap::new();
+
+        let mut current_mapping = MemoryMap::new();
+        let mut current_data = Default::default();
+        for line in reader.lines() {
+            let line = line.map_err(|_| ProcError::Incomplete(Some(path.clone())))?;
+
+            if let Ok(mapping) = MemoryMap::from_line(&line) {
+                hashmap.insert(current_mapping, current_data);
+                current_mapping = mapping;
+                current_data = Default::default();
+            } else {
+                // This is probably an attribute
+                if line.starts_with("VmFlags") {
+                    let flags = line.split_ascii_whitespace();
+                    let flags = flags.skip(1); // Skips the `VmFlags:` part since we don't need it.
+
+                    let flags = flags
+                        .map(|v| match VmFlags::from_str(v) {
+                            None => VmFlags::INVALID,
+                            Some(v) => v,
+                        })
+                        .fold(VmFlags::INVALID, |a, b| a | b);
+
+                    current_data.vm_flags = Some(flags);
+                } else {
+                    let mut parts = line.split_ascii_whitespace();
+
+                    let key = parts.next();
+                    let value = parts.next();
+
+                    if let (Some(k), Some(v)) = (key, value) {
+                        // While most entries do have one, not all of them do.
+                        let size_suffix = parts.next();
+
+                        // Limited poking at /proc/<pid>/smaps and then checking if "MB", "GB", and "TB" appear in the C file that is
+                        // supposedly responsible for creating smaps, has lead me to believe that the only size suffixes we'll ever encounter
+                        // "kB", which is most likely kibibytes. Actually checking if the size suffix is any of the above is a way to
```

Comment:
Looking at [fs/proc/task_mmu.c](https://github.com/torvalds/linux/blob/master/fs/proc/task_mmu.c#L775) concurs with your comment that "kB" is the only used used here, so I'm OK with the code as you've written it.